### DISCUSSION TITLE
why enterprise version docker using Intel QuickSync Encoders not work

### DISCUSSION THREAD

**Discussion by farjar** [ant-media/Ant-Media-Server#6713] [FAQs]
follow the wiki still not work right
https://antmedia.io/docs/guides/advanced-usage/using-intel-quick-sync/

this is my install script
# Update apk
prompt "Update apt..."
apt update

# Install basic requirements
prompt "Install basic requirements..."
apt install cmake git build-essential pkg-config autoconf libtool libdrm-dev libva-dev libx11-dev -y

# Set default Git proxy
prompt "Setting default Git proxy..."
git config --global http.proxy $PROXY_URL
git config --global https.proxy $PROXY_URL

# Install GMMLIB
prompt "Cloning GMMLIB repository..."
git clone https://github.com/intel/gmmlib.git
cd gmmlib/
prompt "Creating build directory for GMMLIB..."
mkdir build && cd build
prompt "Configuring GMMLIB..."
cmake -DCMAKE_BUILD_TYPE=Release -DARCH=64 ..
prompt "Building GMMLIB..."
make
prompt "Installing GMMLIB..."
make install
cd ../..

# Install Libva
prompt "Cloning Libva repository..."
git clone https://github.com/intel/libva.git
cd libva/
prompt "Running autogen for Libva..."
./autogen.sh
prompt "Building Libva..."
make
prompt "Installing Libva..."
make install
cd ..

# Install Intel media driver
prompt "Cloning Intel media driver repository..."
git clone https://github.com/intel/media-driver.git
cd media-driver/
prompt "Creating build directory for media driver..."
mkdir build_media && cd build_media
prompt "Configuring Intel media driver..."
cmake ..
prompt "Building Intel media driver..."
make
prompt "Installing Intel media driver..."
make install
cd ../..

# Install Intel media SDK
prompt "Cloning Intel media SDK repository..."
git clone https://github.com/Intel-Media-SDK/MediaSDK msdk
cd msdk
prompt "Creating build directory for Intel media SDK..."
mkdir build && cd build
prompt "Configuring Intel media SDK..."
cmake ..
prompt "Building Intel media SDK..."
make
prompt "Installing Intel media SDK..."
make install
echo "/opt/intel/mediasdk/lib/" > msdk.conf
prompt "Moving configuration to /etc/ld.so.conf.d/..."
mv msdk.conf /etc/ld.so.conf.d/
prompt "Updating dynamic linker run-time bindings..."
ldconfig
cd ../..

prompt "Configure Ant Media Server..."
echo "settings.encoding.encoderName=h264_qsv" >> /usr/local/antmedia/webapps/WebRTCAppEE/WEB-INF/red5-web.properties
usermod -aG video antmedia

prompt "All installations completed successfully!"

prompt "Restart the Ant Media Server..."
/usr/local/antmedia/antmedia restart

**Comment by farjar** [comment]
maybe provide a officials enterprise docker with Intel QuickSync Encoders installed，that's help me lot

**Reply by yashtandon113** [reply]
Hi @farjar 

Let us review the AMS installation with the Intel Quick Sync Encoder.

Meanwhile, if you want to use the GPU hardware encoder for the streaming via Ant Media Server and Docker container, you can use the Nvidia Nvenc encoder. Check out [this document](https://antmedia.io/docs/guides/clustering-and-scaling/docker/using-nvidia-hardware-based-encoder-on-docker/) for more details.

I hope it helps.

**Reply by farjar** [reply]
use ffmpeg instead, tks
